packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -189,14 +190,8 @@ export function getDeserializePrivateFunction(
     );
     addImportToSpecifier("restClient", runtimeImports, "createRestError");
   } else {
-    const validStatus = [
-      ...new Set(
-        operation.responses
-          .flatMap((r) => r.statusCodes)
-          .filter((s) => s !== "default")
-      )
-    ];
-
+    // Directly using the status code of first response
+    const validStatus = response.statusCodes;
```

Comment:
Directly use the first success status codes.

Reply:
is this because there's is no real case that has multiple responses in typespec?

Reply:
yes.